app/controllers/concerns/foreman/controller/provisioning_templates.rb

```diff
@@ -12,7 +12,21 @@ def load_vars_from_template
 
   def find_global_registration
     template_name = Setting[:default_global_registration_item]
-    @provisioning_template = ProvisioningTemplate.find_by(name: template_name)
+    @provisioning_template = ProvisioningTemplate.unscoped.find_by(name: template_name)
+    @global_registration_vars = global_registration_vars if @provisioning_template
+  end
+
+  def global_registration_vars
+    permitted = Foreman::Plugin.all
+                               .map(&:global_registration_vars)
+                               .flatten.compact.uniq
+    {
+      user: User.current,
+      auth_token: User.current.jwt_token!(expiration: (4 * 60 * 60)),
```

Comment:
This has a side effect. That probably means it should be private.

Reply:
Sorry I don't understand - what do you mean by "it has side effect" and what exactly should be private?

Reply:
Usually `!` indicates a side effect. From the code it looks like does cache the token, so calling it multiple times is probably not a problem. Probably not a real concern then.

Reply:
`4.hours` is probably more readable than `4 * 60 * 60`